switch.go

```diff
@@ -118,30 +82,31 @@ func switchChecker(pass *analysis.Pass, cfg switchConfig, generated boolCache, c
 		sw := n.(*ast.SwitchStmt)
 
 		switchComments := comments.get(pass.Fset, file)[sw]
-		uDirectives := userDirectives(switchComments)
-		if !cfg.explicit && directivesIncludes(uDirectives, ignoreComment) {
+		uDirectives := parseDirectiveSet(switchComments)
+
+		if !cfg.explicit && uDirectives.hasDirective(ignoreDirective) {
 			// Skip checking of this switch statement due to ignore
 			// comment. Still return true because there may be nested
 			// switch statements that are not to be ignored.
 			return true, resultIgnoreComment
 		}
-		if cfg.explicit && !directivesIncludes(uDirectives, enforceComment) {
+		if cfg.explicit && !uDirectives.hasDirective(enforceDirective) {
 			// Skip checking of this switch statement due to missing
 			// enforce comment.
 			return true, resultNoEnforceComment
 		}
 		requireDefaultCase := cfg.defaultCaseRequired
-		if directivesIncludes(uDirectives, ignoreDefaultCaseRequiredComment) {
+		if uDirectives.hasDirective(ignoreDefaultCaseRequiredDirective) {
 			requireDefaultCase = false
 		}
-		if directivesIncludes(uDirectives, enforceDefaultCaseRequiredComment) {
+		if uDirectives.hasDirective(enforceDefaultCaseRequiredDirective) {
 			// We have "if" instead of "else if" here in case of conflicting ignore/enforce directives.
```

Comment:
I wonder whether it would make more sense for the analyzer to raise an error in this case. I don't foresee it happening much in practice though.

I'm more inclined to believe typos like `//exhaustive:enfocre` should result in errors.